code/Control/UserDefinedFormController.php

```diff
@@ -562,8 +563,26 @@ function (){
             {$holder}.{$rule['opposite']}.trigger('{$rule['holder_event_opposite']}');
         }
     });
+EOS;
+            if ($isFormStep) {
+                // Hide the step jump button if the FormStep has is initially hidden.
+                // This is particularly important beacause the next/prev page buttons logic is controlled by
+                // the visibility of the FormStep buttons
+                // The HTML for the FormStep buttons is defined in UserFormProgress.ss
+                $id = str_replace('#', '', $target);
```

Comment:
I'm fine with using `str_replace`. 

I find this syntax is a little bit more expressive.

```suggestion
                $id = ltrim($target, '#');
```